README.md

```diff
@@ -46,52 +46,92 @@ For each module, the registry contains a JSON file with the following structure:
+        "functionName": "<the function's name in the contract (string)>",
+        "label": "<Human readable name for the function, for display purpose (string)>",
+        "description": "<function's descruption (string)>",
+        "primary": "<optional field, indicates that this is the role's primary fucntion, for display purpose (boolean)>"
+        "args": [
+          {
+            "name": "<argument's human readable name, for display purpose (string)>",
+            "description": "<argument's description (string)>",
+            "type": "<argument's solidity type>",
+            "displayType": "<arguments's type of content, for display purpose>",
+            "optional": "<optional field, setting to 'true' indicates that this input is optional>"
+          }
+        ]
+     }
+  ]
   "abi": "<module's contract ABI>"
 }
 ```
 
-(Note that arrays in the object above contain one example entry).
-
-Here are some notes on its expected structure:
-
-- The "name" property represents the name of the module.
-- The "details" property is structured as an array of strings. Each array entry represents a paragraph (which will be rendered in a UI).
-- The "links" property should have at least one link to the module's source code.
-- The "parameters" property is used in order to dynamically fetch and display data from module instances.
-  - The "label" property is used to display the name/description of each parameter.
-  - The "functionName" property represents the name of the function from which the parameter should be retrieved. The function should be a view/pure function with no input and one output (not a tuple, but can be an array).
-  - The "displayType" property is used in order to display a proper UI component for the parameter. For example, displaying a date for a parameter representing a timestamp. The supported types are currently:
-    - "default" - Infers automatically the UI component for the value
-    - "timestamp" - Unix timestamp, will be presented as a Date component
-    - "hat" - hat component
-    - "token" - Token component
-    - "seconds" - Amount of time denominated in seconds
-    - "amountWithDecimals" - For token amounts, takes into account the token's decimals
-- There should be at least one field on the "type" property which is set to "true". A module might serve as more than one type, in which case there will be more than one field set to "true".
-- For each chain provided in "deployments", there should be a deployed implementation contract with an address matching the provided "implementationAddress" property and with an ABI
-  matching the "abi" property.
-- For both the immutable and mutable arguments, their order should match the order of the arguments as provided to the module.
-- The "example" fields will be used to automatically test the module's deployment.
+_Note that arrays in the object above contain one example entry._
+
+**Here are some useful notes on its expected structure:**
```

Comment:
I think this would be more readable with clearer formatting. Maybe separate into sections (with markdown headers) for each property. Something like...

```
### `parameters` 
This property is used in order to dynamically fetch and display data from module instances. Using this property, the module creator can choose which module fields are relevant for display.

  - `label` - Used to display the name/description of each parameter.
  - `functionName` - Name of the function from which the parameter should be retrieved. The function should be a view/pure function with no inputs and only one output (an array is also considered as one output).
  - `displayType` - Used in order to display a proper UI component for the parameter. For example, displaying a date for a parameter representing a timestamp. The known supported types are currently:
    - `default` - Infers automatically the UI component for the value.
    - `timestamp` - a value which represnts a Unix timestamp.
    - `hat` - A `uint256` value which represents a hat ID.
    - `token` - An `address` value of a token contract.
    - `seconds` -A value which represents time denominated in seconds.
    - `amountWithDecimals` - A value which represnts a token amount, allows for taking into account the token's decimals.
```